### PR TITLE
FlowFolder

### PR DESCRIPTION
**Pull request by dolymood**
When we upload a folder, may be we want to show the folder rather than the files in it.

So i created a new branch in https://github.com/dolymood/flow.js/tree/folder.I added `FlowFolder` constructor and the `parsedFiles` property to `Flow` object.

Maybe it is a good idea.

Or support an option like `showFolder` in the `Flow` options.

### CODE REVIEW

**Comment by AidasK** [comment]
This looks like a big feature and well done, but I don't get the idea why it is needed.

**Comment by dolymood** [comment]
A `folder` can be treated as an entry.
You can show a `folder` item in your page, then you can `pause`, `resume` or `cancel` the files by handle the `folder` item. This will be look like a "real" root directory which contains some files and have some `file` siblings.

Or the `FolowFolder` will be a plugin of flow.js and i will modify my files. :smile:

**Comment by blazedd** [comment]
Is this project dead or is the author ignoring any contributions? This is well over a year old and would be a really nice feature :(

**Comment by AidasK** [comment]
I am not sure we need this feature. This pull request contains so many code changes, it would require some decent testing. For now, only two people requested this, so it is a low priority. It would be better to have this as a plugin

**Comment by johaven** [comment]
It would be nice to have this feature, i often need to send folder with a lot of files (more than 1000 files) and it's difficult to follow the upload status of all these files.
I have no solution to track the files uploaded from one specific folder (i must use global flow progress which is no representative of my folder's files ...).
I must track error per file, connect to signal complete and filter files which are a parts of my folder ...
Display side, i use a subclass to freeze variables that will be modified during the next upload (FlowObj uploaded size, remaining time ...).
In short, sending folders is not optimal and could be much simpler.

**Comment by AidasK** [comment]
Currently I don't have time to merge this feature. I gave a quick peek and I think it needs some performance improvements. And what about angular? It might not be compatible. As @dolymood has mentioned, these changes can be moved to a separate plugin. I could merge it as a separate file. It does not need to be tested or perfect.

**Comment by johaven** [comment]
Just one question a little bit off subject, I'm trying to see if I can create a second instance of the flow to dedicate it to a folder when it is detected by the first instance.
Apparently this is not possible, and I do not understand why! The Flow instance is a singleton?
Example:
```javascript
import * as Flow from '@flowjs/flow.js'

const a = new Flow({...})
const b = new Flow({...})
b.custom_attribute = "test"
console.log(a.custom_attribute)
-> "test"
```
